### PR TITLE
Removing the HTTP_Request2 package.

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

**Comment by sergey-shandar** [comment]
Partly #797

File: tests/framework/BlobServiceRestProxyTestBase.php [hunk 1]
```diff
@@ -25,7 +25,8 @@
 
 namespace Tests\framework;
```
Comment (yaqiyang) [inline tests/framework/BlobServiceRestProxyTestBase.php]: No need to fix the storage test files. You can just get rid of them.

Comment (sergey-shandar) [inline tests/framework/BlobServiceRestProxyTestBase.php]: I will :-) May be in next PR.

**Comment by yaqiyang** [comment]
also, I would prefer to squash the commits since you have 93 of them on this branch.

File: src/Common/Internal/Filters/AuthenticationFilter.php [hunk 2]
```diff
@@ -81,12 +82,12 @@ public function handleRequest(IHttpClient $request)
     /**
      * Does nothing with the response.
      *
-     * @param IHttpClient             $request  HTTP channel object.
-     * @param \HTTP_Request2_Response $response HTTP response object.
+     * @param IHttpClient       $request  HTTP channel object.
+     * @param ResponseInterface $response HTTP response object.
```
Comment (yaqiyang) [inline src/Common/Internal/Filters/AuthenticationFilter.php]: Ignore this if the definition is from Guzzle.

Comment (sergey-shandar) [inline src/Common/Internal/Filters/AuthenticationFilter.php]: ResponseInterface is defined in PSR 7 http://www.php-fig.org/psr/psr-7/ and it has all functionality that we need. I don't see a point to create a wrap on top of it. But we may wrap in the future if we have a good reason for it.

**Comment by sergey-shandar** [comment]
NP. I can squash it because I don't have branches which depends on this one.

**Pull request merged**